src/plugins/unified_search/public/filter_bar/filter_editor/filter_editor.tsx

```diff
@@ -143,42 +144,69 @@ export interface FilterEditorComponentProps {
   suggestionsAbstraction?: SuggestionsAbstraction;
   docLinks: DocLinksStart;
   filtersCount?: number;
+  dataViews: DataViewsContract;
 }
 
 export type FilterEditorProps = WithEuiThemeProps & FilterEditorComponentProps;
 
 interface State {
+  indexPatterns: DataView[];
   selectedDataView?: DataView;
   customLabel: string | null;
   queryDsl: string;
   isCustomEditorOpen: boolean;
   localFilter: Filter;
+  isLoadingDataView?: boolean;
 }
 
 class FilterEditorComponent extends Component<FilterEditorProps, State> {
   constructor(props: FilterEditorProps) {
     super(props);
-    const dataView = this.getIndexPatternFromFilter();
+    const dataView = getIndexPatternFromFilter(props.filter, props.indexPatterns);
     this.state = {
+      indexPatterns: props.indexPatterns,
       selectedDataView: dataView,
       customLabel: props.filter.meta.alias || '',
-      queryDsl: this.parseFilterToQueryDsl(props.filter),
+      queryDsl: this.parseFilterToQueryDsl(props.filter, props.indexPatterns),
       isCustomEditorOpen: this.isUnknownFilterType() || !!this.props.filter?.meta.isMultiIndex,
       localFilter: dataView ? merge({}, props.filter) : buildEmptyFilter(false),
+      isLoadingDataView: !Boolean(dataView),
     };
   }
 
   componentDidMount() {
-    const { localFilter, queryDsl, customLabel } = this.state;
+    const { localFilter, queryDsl, customLabel, selectedDataView } = this.state;
     this.props.onLocalFilterCreate?.({
       filter: localFilter,
       queryDslFilter: { queryDsl, customLabel },
     });
     this.props.onLocalFilterUpdate?.(localFilter);
+    if (!selectedDataView) {
+      const dataViewId = this.props.filter.meta.index;
+      if (!dataViewId) {
+        this.setState({ isLoadingDataView: false });
+      } else {
+        try {
+          this.props.dataViews.get(dataViewId).then((dataView) => {
```

Comment:
Maybe this needs a `.catch` statement instead of wrapping this block inside of a `try`/`catch`? (Since it relies on promises, not async/await)